### PR TITLE
fix(sdk): fix hang after failed wandb.init

### PR DESCRIPTION
**Pull request by raubitsj**
Description
-----------
Add a "started" state to the streams object in `wandb service`

Testing
-------
yea test added to test the init fail case

Checklist
-------
- [x] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/master/CONTRIBUTING.md#conventional-commits)

### CODE REVIEW

File: wandb/sdk/service/streams.py [hunk 1]
```diff
@@ -55,8 +55,10 @@ class StreamRecord:
     _iface: InterfaceRelay
     _thread: StreamThread
     _settings: SettingsStatic  # TODO(settings) replace SettingsStatic with Setting
+    _active: bool
```
Comment (kptkin) [inline wandb/sdk/service/streams.py:58]: nit: do you mind making it some kind of an object, either enum or something else (I assume we will keep other states)

File: wandb/sdk/service/streams.py [hunk 2]
```diff
@@ -285,6 +288,8 @@ def _finish_all(self, streams: Dict[str, StreamRecord], exit_code: int) -> None:
         # and jupyter is disabled if at least single stream's setting set `_jupyter` to false
         exit_handles = []
         for stream in streams.values():
```
Comment (kptkin) [inline wandb/sdk/service/streams.py:290]: nit: what do you think of making streams some iterable class that will return only the valid streams?

**Review by kptkin — APPROVED** [review]
Yikes... this is a bad one... thanks for the fix! Totally makes sense

File: wandb/sdk/service/streams.py (hunk 1 above)
Comment (raubitsj) [reply wandb/sdk/service/streams.py]: I kept it as a boolean as it isnt clear what the other states will be and whether they will be independent..
i did rename it to be more clear, and added an explicit "stream_start" action.

File: wandb/sdk/service/streams.py (hunk 2 above)
Comment (raubitsj) [reply wandb/sdk/service/streams.py]: Added a staticmethod for started streams

**Comment by raubitsj** [comment]
This is now part of the resiliency PR

**Pull request closed without merging**